src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -459,17 +459,21 @@ CHIP_ERROR DiscoveryImplPlatform::FinalizeServiceUpdate()
     return ChipDnssdFinalizeServiceUpdate();
 }
 
-CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId, Inet::IPAddressType type)
+CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId, Inet::IPAddressType type,
+                                                Resolver::CacheBypass dnssdCacheBypass)
 {
     ReturnErrorOnFailure(InitImpl());
 
 #if CHIP_CONFIG_MDNS_CACHE_SIZE > 0
-    /* see if the entry is cached and use it.... */
-    ResolvedNodeData nodeData;
-    if (sDnssdCache.Lookup(peerId, nodeData) == CHIP_NO_ERROR)
+    if (dnssdCacheBypass == Resolver::CacheBypass::Off)
```

Comment:
Rather than bypassing the cache for this one single call, shouldn't we be evicting the entry or somehow marking it stale?

Reply:
Yes, you are right, but based on that comment: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/dnssd/DnssdCache.h#L56 I think that there is already a plan to do that. Do you think is could be handled outside of this PR?

Reply:
> Do you think it could be handled outside of this PR? 

That TODO is to handle eviction when the cache is full.

To me this is a different problem, and I am of the opinion that what this PR is doing just isn't correct.  The new code attempts to update records when it determines they are stale, but doesn't evict the stale records.  Why are we keeping a stale record?  Do we intend to revert to the old record and try it again?

If not, I think this PR should implement a DiscoveryImplPlatform::EvictRecord(const PeerId & peerId) method that calls DnssdCache::Delete(PeerId peerId).  It would not be more difficult than what this PR is doing and seems more correct.